jerry-core/jerry.c

```diff
@@ -519,6 +519,103 @@ jerry_value_is_undefined (const jerry_value_t value) /**< api value */
 } /* jerry_value_is_undefined */
 
 /**
+ * Check if the specified feature is enabled.
+ *
+ * @return true  - if the specified feature is enabled,
+ *         false - otherwise.
+ */
+bool jerry_is_feature_enabled (const jerry_feature_t feature)
+{
+  switch (feature)
+  {
+    case JERRY_FEATURE_ERROR_MESSAGES:
```

Comment:
Apart from the incorrect style the idea is not bad

    switch (feature)
    {
    #ifdef FEATURE1
       case FEATURE1:
    #endif /* FEATURE1 */
    #ifdef FEATURE2
       case FEATURE2:
    #endif /* FEATURE2 */
       {
         return true;
       }
       default:
       {
         return false;
       }
    }

This could work as long as at least one feature is enabled.

Or just using a conditional form:

    return (false
    #ifdef FEATURE1
            || feature == FEATURE1
    #endif /* FEATURE1 */
    #ifdef FEATURE2
            || feature == FEATURE2
    #endif /* FEATURE2 */
            || false);